src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -28,8 +28,9 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
-    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)_AssemblyInfo.cs</AssemblyInfoFile>
-    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)_AssemblyInfo.vb</AssemblyInfoFile>
+    <AssemblyInfoFileDir>$(BaseIntermediateOutputPath)$(MSBuildProjectName)\</AssemblyInfoFileDir>
```

Comment:
What about the other version files generated like versions.h for the native stuff. Does that also need to be in a neutral location?